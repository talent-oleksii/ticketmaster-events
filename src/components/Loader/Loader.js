// Packages
import React from "react";
import styled, { keyframes } from 'styled-components';

// Styles
const rotate = keyframes`
    from {
        transform: rotate(0deg);
    }

    to {
        transform: rotate(360deg);
    }
`;

const LoaderItem = styled.div`
    margin: 0 auto;
    vertical-align: center;
    border: 8px solid #f3f3f3;
    border-radius: 50%;
    border-top: 8px solid #3498db;
    width: 60px;
    height: 60px;
    -webkit-animation: ${rotate} 2s linear infinite; /* Safari */
    animation: ${rotate} 2s linear infinite;
`


export class Loader extends React.Component {
  render() {
    return(
        <div>
            <LoaderItem></LoaderItem>
        </div>
    );
  }
}


export default Loader;
