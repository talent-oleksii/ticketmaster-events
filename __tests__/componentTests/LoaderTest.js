// Packages
import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

// Components
import Loader from '../../src/components/Loader/Loader';


configure({ adapter: new Adapter() });

describe('Loader Component', () => {
    it('Should render without errors', () => {
        const wrapper = shallow(<Loader />);
        expect(wrapper.exists()).toBe(true)
    });

    test('Snaphot testing', () => {
        const wrapper = shallow(
            <Loader />
        );
        expect(wrapper).toMatchSnapshot();
    })
})
