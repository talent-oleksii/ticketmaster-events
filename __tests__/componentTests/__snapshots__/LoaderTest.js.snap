// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Component Snaphot testing 1`] = `ShallowWrapper {}`;
